tests/main_test.ts

```diff
@@ -191,6 +191,15 @@ Deno.test("/foo/:path*", async () => {
   assert(body.includes("bar/baz"));
 });
 
+Deno.test("decodes parameters", async () => {
+  const resp = await router(new Request("https://fresh.deno.dev/this has spaces"));
```

Comment:
You may have considered this already but it may be a good idea to put some query params 
Perhaps: `https://fresh.deno.dev/this has spaces?a=1&b=2&c=3`

Reply:
This issue isn't about query parameters, but path parameters, which the test does assert correctly.